src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs

```diff
@@ -128,6 +128,66 @@ public static void HashBytes(HashAlgorithm hashAlgorithm, byte[] bytes)
 #endif
         }
 
+
+        /// <summary>
+        /// Read bytes from a BinaryReader and hash them with a given HashAlgorithm wrapper and stop when the provided position
+        /// is the current position of the BinaryReader's base stream. It does not hash the byte in the provided position.
+        /// </summary>
+        /// <param name="reader">Read bytes from this stream</param>
+        /// <param name="hashFunc">HashAlgorithm wrapper used to hash contents cross platform</param>
+        /// <param name="position">Position to stop copying data</param>
+        internal static void ReadAndHashUntilPosition(BinaryReader reader, Sha512HashFunction hashFunc, long position)
```

Comment:
Why use a `Sha512HashFunction` instead of a `HashAlgorithm` or other to make the method be able to hash for any algorithm passed instead of just sha512?

Reply:
Should we add a comment or something here so that we can revisit this method once we add the work for signing in dotnet core? So we can just reuse this method whenever we add this support.